nextest-runner/src/test_list.rs

```diff
@@ -86,6 +85,11 @@ impl<'g> RustTestArtifact<'g> {
                         if artifact.target.name != package.name() {
                             binary_id.push_str("::");
                             binary_id.push_str(&artifact.target.name);
```

Comment:
Hmm, what happens if there's multiple `bin` targets within a crate, each with their own tests?

I think the best scheme for this is probably something like:
`<crate-name>` for the lib crate (unit tests)
`<crate-name>::<bin-name>` for test targets
`<crate-name>::<bin/example>/<bin-name>` for other targets, e.g in the included test, `nextest-tests::bin/nextest-tests`

What do you think? Happy to hear suggestions as well.

(And could you add more tests/examples around this? Thanks!)

Reply:
Aha, you're right - the problem is more general than I thought. I think your naming scheme makes sense - I'll give it a go.

Reply:
So I think this works, with one small exception: if the name of the lib is different from the package, it's not reflected in the binary_id now. It's still unique and un-ambiguous, but we could also use
`<crate-name>::lib/<lib-name>` in that case if you think it'd be confusing not seeing it.

Reply:
> if the name of the lib is different from the package, it's not reflected in the binary_id now.

Good catch, I keep forgetting about this use case. I wouldn't address this personally.